src/gsy_e/gsy_e_core/redis_connections/redis_area_market_communicator.py

```diff
@@ -103,59 +153,64 @@ def __init__(self, is_enabled):
             self.channel_callback_dict = {}
             self.aggregator = AggregatorHandler(self.redis_db)
 
-    def sub_to_channel(self, channel, callback):
+    def sub_to_channel(self, channel: str, callback: Callable):
         if not self.is_enabled:
             return
         self.pubsub.subscribe(**{channel: callback})
 
-    def sub_to_multiple_channels(self, channel_callback_dict):
+    def sub_to_multiple_channels(self, channel_callback_dict: Dict):
         if not self.is_enabled:
             return
         self.pubsub.subscribe(**channel_callback_dict)
 
     def start_communication(self):
+        """Start pubsub thread."""
         if not self.is_enabled:
             return
         if not self.pubsub.subscribed:
             return
         thread = self.pubsub.run_in_thread(sleep_time=0.1, daemon=True)
-        log.debug(f"Started ExternalConnectionCommunicator thread for "
-                  f"multiple channels: {thread}")
+        log.debug("Started ExternalConnectionCommunicator thread for multiple channels: %s",
+                  thread)
         self.thread = thread
 
     def sub_to_aggregator(self):
+        """Subscribe to aggregator channels."""
         if not self.is_enabled:
             return
         channel_callback_dict = {
-            f'external/{gsy_e.constants.CONFIGURATION_ID}/aggregator/*/batch_commands':
+            f"external/{gsy_e.constants.CONFIGURATION_ID}/aggregator/*/batch_commands":
                 self.aggregator.receive_batch_commands_callback,
-            'aggregator': self.aggregator.aggregator_callback
+            "aggregator": self.aggregator.aggregator_callback
         }
         self.pubsub.psubscribe(**channel_callback_dict)
 
     def approve_aggregator_commands(self):
+        """Wrapper for calling approve_batch_commands."""
         if not self.is_enabled:
             return
         self.aggregator.approve_batch_commands()
 
     def publish_aggregator_commands_responses_events(self):
+        """Wrapper for publishing aggregator command responses and events."""
         if not self.is_enabled:
             return
         self.aggregator.publish_all_commands_responses(self)
         self.aggregator.publish_all_events(self)
 
 
-class BlockingCommunicator(RedisCommunicator):
-    def __init__(self):
-        super().__init__()
-        self.lock = Lock()
+class RQExternalConnectionCommunicator(ExternalConnectionCommunicator, RQResettableCommunicator):
+    """Communicator for sending messages using redis queue including utils for aggregator."""
```

Comment:
Why don't we just override `publish_json` here and avoid mixins? Something like:
```python
class RQExternalConnectionCommunicator(ExternalConnectionCommunicator):
    def publish_json(self, channel: str, data: Dict) -> None:
        ...
```
Then we could remove the `RQResettableCommunicator` class. What do you think?

Reply:
The `ExternalConnectionCommunicator` and `RQExternalConnectionCommunicator` are inheriting from the `RQResettableCommunicator`, in order to remove the `RQResettableCommunicator` we would need to move all code from there into the `ExternalConnectionCommunicator`. But they have different implementations of e.g. `sub_to_response`. 
I did not want to duplicate code which is why I created the RQExternalConnectionCommunicator that inherits all from ExternalConnectionCommunicator and overwrites the `publish_json` with the  `RQResettableCommunicator.publish_json`
Let me know if there is a more cleaner way. Thanks

Reply:
I'm not sure if I'm missing something. `ExternalConnectionCommunicator` and `RQExternalConnectionCommunicator` inherit from the same class (`ResettableCommunicator`). `RQResettableCommunicator` redefines only one method (`publish_json`). My proposal is to delete `RQResettableCommunicator` and directly redefine `publish_json` inside of `RQExternalConnectionCommunicator` (as in the snippet I put in the comment above).
`RQExternalConnectionCommunicator` would still inherit from `ExternalConnectionCommunicator`, so we wouldn't have to duplicate code.
Sorry if I'm not seeing something!

Reply:
I'm not sure if I'm missing something. `ExternalConnectionCommunicator` and `RQResettableCommunicator` inherit from the same class (`ResettableCommunicator`). `RQResettableCommunicator` redefines only one method (`publish_json`). My proposal is to delete `RQResettableCommunicator` and directly redefine `publish_json` inside of `RQExternalConnectionCommunicator` (as in the snippet I put in the comment above).
`RQExternalConnectionCommunicator` would still inherit from `ExternalConnectionCommunicator`, so we wouldn't have to duplicate code.
Sorry if I'm not seeing something!

Reply:
I understood, what you were proposing, but we need the `RQResettableCommunicator` for the myco communication. `ExternalConnectionCommunicator` and `RQExternalConnectionCommunicator` is used for the aggregator communication and overwrite methods, just like `sub_to_channel` from their baseclass `ResettableCommunicator`. 
Sorry If I misunderstood something here. Please let me know @fievelk

Reply:
Ahh ok, here's what I hadn't seen! I see your point now. I usually try to avoid mixins, that's probably why I got suspicious. Thanks!